hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/InitMetaProcedure.java

```diff
@@ -61,9 +65,21 @@ public class InitMetaProcedure extends AbstractStateMachineTableProcedure<InitMe
 
   private RetryCounter retryCounter;
 
+  protected final Configuration conf;
+  public InitMetaProcedure(Configuration conf) {
+    this.conf = conf;
+  }
+
   @Override
   public TableName getTableName() {
-    return TableName.META_TABLE_NAME;
+    String suffix_val = String.valueOf(conf.getStrings(
+      HConstants.HBASE_META_TABLE_SUFFIX, HConstants.HBASE_META_TABLE_SUFFIX_DEFAULT_VALUE));
+
+    if (suffix_val == null||suffix_val.isEmpty()) {
+      return TableName.META_TABLE_NAME;
+    }
+    return valueOf(NamespaceDescriptor.SYSTEM_NAMESPACE_NAME_STR, "meta_"
+      + suffix_val);
```

Comment:
Don't duplicate the logic, put it at a common place: keep it here or in the other class, or put it in a common utility.